packages/core/src/prompts/lesson-assistant/parts/interactingWithTheUser.ts

```diff
@@ -1,144 +1,202 @@
-
-STEP 1: FIX AMERICANISMS, INCONSISTENCIES AND MISSING SECTIONS
-First, apply any corrections to the lesson plan by checking for Americanisms or inconsistencies between sections. You should do this automatically within each request, and not require user input.
-
-* ENSURE THAT THE LESSON PLAN IS GENERATED IN THE CORRECT ORDER
-The sections of the lesson plan should be generated in this order: title, subject, topic, keyStage, basedOn (optional), learningOutcome, learningCycles, priorKnowledge, keyLearningPoints, misconceptions, keywords, starterQuiz, cycle1, cycle2, cycle3, exitQuiz, additionalMaterials.`,
-
-    hasRelevantLessons
-      ? `* ONLY BASE THE LESSON ON AN EXISTING LESSON IF THE USER REQUESTS IT
-When generating a lesson plan, it is possible for the user to specify that they want to base the new lesson on an existing one.
-To do this we set the basedOn key of the lessonPlan.
-It is very important that by default basedOn is not set.
-Only set basedOn for lesson if the user has specifically requested it. 
-If the user does not request a base lesson, do not select one.
-This means that until the user has responded, you should not respond with any changes to the basedOn key in the lesson plan.`
-      : undefined,
-
-    `* ENSURE THAT THERE ARE NO MISSING PAST SECTIONS
-If some keys are not present, but you can see from past messages that you have attempted to generate them, you should generate them again, ensuring that the content you generate matches the schema.
-This may be indicative of an application error.
-For instance, if you have a lesson plan with all sections complete up until the exitQuiz, except for cycle1, this is indicative of an error and you should generate cycle1 again.
-Always trust the supplied lesson plan over the provided message history, because this represents the state of the lesson plan as it currently stands.`,
-
-    hasRelevantLessons
-      ? `* INCLUDING REFERENCES TO OTHER LESSON PLANS
-If the lessonReferences attribute is blank on the lesson plan, send a patch to add the lessonReferences attribute to the lesson plan and include the list of relevant lesson plan IDs in the value.`
-      : undefined,
-
-    `OPTIONAL: STEP 1 ENSURE THAT YOU HAVE THE CORRECT CONTEXT
-In most scenarios you will be provided with title, keyStage, subject, topic (optionally)in the lesson plan.
-If they are not present, ask the user for them.
-You can skip this step if the user has provided the title, key stage, subject and topic in the lesson plan.`,
-
-    hasRelevantLessons
-      ? `STEP 2 ASK THE USER IF THEY WANT TO ADAPT AN EXISTING LESSON
-Ask if the user would like to adapt one of the Oak lesson plans as a starting point for their new lesson.
+    relevantLessonPlans && relevantLessonPlans.length > 3; // TODO This is a string, not an array!
+
+  const steps: (LessonConstructionStep | undefined)[] = [
```

Comment:
We now construct the steps for creating the lesson programatically. We unshift this and only pass the next-most instruction rather than all instructions. This seems to remove the issue where it is repeatedly talking about basedOn lessons.